src/gtc/oir.py

```diff
@@ -145,19 +145,68 @@ class Interval(LocNode):
     start: AxisBound
     end: AxisBound
 
+    @root_validator
+    def check(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        start, end = values["start"], values["end"]
+        if start.level == common.LevelMarker.END and end.level == common.LevelMarker.START:
+            raise ValueError("Start level must be smaller or equal end level")
+        if start.level == end.level and start.offset > end.offset:
+            raise ValueError("Start offset must be smaller or equal end offset")
```

Comment:
I don't know which is correct, but text and condition don't match `>=` or `>`?
Also maybe add "If start and end level are equal, start offset..."

Reply:
I think it does match (`start.offset > end.offset` as used in the error condition is the negation of the required condition `start.offset <= end.offset`). But maybe `not (start.offset <= end.offset)` would be more readable?

Reply:
Interval ranges are exclusive, so shouldn't it be
 `start.level == end.level and start.offset >= end.offset` (or `not (start.offset < end.offset)`)
 Except if we want to allow empty intervals to be represented.

Reply:
No idea if we need empty intervals, I just cared about ordered start and end. I disabled empty intervals now, let’s see if the tests pass…